ush/generate_FV3LAM_wflow.py

```diff
@@ -506,24 +507,23 @@ def generate_FV3LAM_wflow(
     #
     # -----------------------------------------------------------------------
     #
-    # Call the set_namelist.py script to create a new FV3 namelist file (full
-    # path specified by FV3_NML_FP) using the file FV3_NML_BASE_SUITE_FP as
-    # the base (i.e. starting) namelist file, with physics-suite-dependent
-    # modifications to the base file specified in the yaml configuration file
-    # FV3_NML_YAML_CONFIG_FP (for the physics suite specified by CCPP_PHYS_SUITE),
-    # and with additional physics-suite-independent modifications specified
-    # in the variable "settings" set above.
+    # Create a new FV3 namelist file
     #
     # -----------------------------------------------------------------------
     #
-    args=[ "-n", FV3_NML_BASE_SUITE_FP,
-           "-c", FV3_NML_YAML_CONFIG_FP, CCPP_PHYS_SUITE,
-           "-u", settings_str,
-           "-o", FV3_NML_FP,
-          ]
-    if not debug:
-        args.append("-q")
-    set_namelist(args)
+
+    physics_cfg = get_yaml_config(FV3_NML_YAML_CONFIG_FP)
+    base_namelist = get_nml_config(FV3_NML_BASE_SUITE_FP)
+    base_namelist.update_values(physics_cfg[CCPP_PHYS_SUITE])
+    base_namelist.update_values(settings)
+    for sect, values in base_namelist.copy().items():
+        if not values:
+            del base_namelist[sect]
+            continue
+        for k, v in values.copy().items():
+            if v is None:
+                del base_namelist[sect][k]
+    base_namelist.dump(FV3_NML_FP)
```

Comment:
The `set_namelist` tool handled all of the steps now included here with the help of the uwtools API.